dali/pipeline/workspace/workspace.h

```diff
@@ -247,23 +243,32 @@ class WorkspaceBase : public ArgumentWorkspace {
   OutputType<GPUBackend> SharedGPUOutput(int idx) {
     DALI_ENFORCE_VALID_INDEX(idx, output_index_map_.size());
     auto tensor_meta = output_index_map_[idx];
-    DALI_ENFORCE(!tensor_meta.first, "Output with given "
+    DALI_ENFORCE(tensor_meta.storage_device == StorageDevice::GPU, "Output with given "
         "index does not have the calling backend type (GPUBackend)");
-    return gpu_outputs_[tensor_meta.second];
+    return gpu_outputs_[tensor_meta.index];
   }
 
  protected:
+  struct InOutMeta {
+    StorageDevice storage_device;
+    int index;
```

Comment:
It'd be nice to provide a comment, what is it ;)

Reply:
Done